src/Qir/Controller/Executable/ClangClient.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Utility;
+
+namespace Microsoft.Quantum.Qir.Executable
+{
+    public class ClangClient : IClangClient
+    {
+        private const string LinkFlag = " -l ";
+        private readonly ILogger logger;
+
+        public ClangClient(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        public async Task CreateExecutableAsync(string[] inputFiles, string[] libraries, string libraryPath, string outputPath)
```

Comment:
Related to this [comment](https://github.com/microsoft/qsharp-runtime/pull/620/files#r610263153), this method would need a new parameter (`string includePath`) which should be added to the arguments passed to `clang` using the `-I` option.